daemon/src/model.rs

```diff
@@ -600,6 +602,70 @@ impl Timestamp {
     }
 }
 
+/// Funding rate per SETTLEMENT_INTERVAL
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct FundingRate(Decimal);
+
+impl FundingRate {
+    pub fn new(rate: Decimal) -> Result<Self> {
+        anyhow::ensure!(
+            rate.is_sign_positive(),
+            "Negative funding rates (ie. paid by the maker) are not supported yet"
+        );
+        anyhow::ensure!(
+            rate <= Decimal::ONE,
+            "Funding rate can't be higher than 100%"
+        );
+
+        Ok(Self(rate))
+    }
+
+    pub fn as_dec(&self) -> Decimal {
```

Comment:
Nit: Rust guidelines recommend `as_` only for `&` -> `&` conversions. We are decreasing abstraction here I think the correct API as per https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv would be `pub fn to_decimal(self) -> Decimal` given that this type is `Copy` :)

Reply:
thanks, bookmarked! spent a few minutes trying to get clippy lint working for it, but I think it's broken :/